From a27bc4778ef4579a954846cb75435226655ad31a Mon Sep 17 00:00:00 2001
From: Michal Sekletar <msekleta@redhat.com>
Date: Fri, 30 May 2014 18:20:16 +0200
Subject: [PATCH] localed: consider an unset model as a wildcard

(cherry picked from commit 387066c2e5bda159201896b194711965b52f34a9)

Conflicts:
        src/locale/localed.c

Resolves: #903776
---
 src/locale/localed.c | 13 +++++++------
 1 file changed, 7 insertions(+), 6 deletions(-)

diff --git a/src/locale/localed.c b/src/locale/localed.c
index b9b98f4..a93b309 100644
--- a/src/locale/localed.c
+++ b/src/locale/localed.c
@@ -871,15 +871,16 @@ static int find_legacy_keymap(char **new_keymap) {
                         }
                 }
 
-                if (matching > 0 &&
-                    streq_ptr(state.x11_model, a[2])) {
-                        matching++;
-
-                        if (streq_ptr(state.x11_variant, a[3])) {
+                if (matching > 0) {
+                        if (isempty(state.x11_model) || streq_ptr(state.x11_model, a[2])) {
                                 matching++;
 
-                                if (streq_ptr(state.x11_options, a[4]))
+                                if (streq_ptr(state.x11_variant, a[3])) {
                                         matching++;
+
+                                        if (streq_ptr(state.x11_options, a[4]))
+                                                matching++;
+                                }
                         }
                 }
 
